prov/sockets/src/sock_progress.c

```diff
@@ -2089,6 +2098,20 @@ static int sock_pe_progress_tx_entry(struct sock_pe *pe,
 	int ret;
 	struct sock_conn *conn = pe_entry->conn;
 
+	if (sock_comm_is_disconnected(pe_entry)) {
+		SOCK_LOG_DBG("conn disconnected: removing fd from pollset\n");
+		if (pe_entry->conn->sock_fd != -1) {
```

Comment:
Please update the logic as it's in ```sock_pe_progress_rx_pe_entry```. There was a recent fix that updated this logic.